plugins/octant/cmd/antrea-octant-plugin/traceflow.go

```diff
@@ -88,139 +95,167 @@ func getDstType(tf *crdv1alpha1.Traceflow) string {
 	return ""
 }
 
-// actionHandler handlers clicks and actions from "Start New Trace" and "Generate Trace Graph" buttons.
+// actionHandler handlers clicks and actions from "Start New Trace", "START NEW LIVE-TRAFFIC TRACE" and "Generate Trace Graph" buttons.
 func (p *antreaOctantPlugin) actionHandler(request *service.ActionRequest) error {
 	actionName, err := request.Payload.String("action")
 	if err != nil {
-		log.Printf("Failed to get input at string: %s", err)
+		log.Printf("Failed to get input at string: %s\n", err)
 		return nil
 	}
-
 	switch actionName {
-	case addTfAction:
-		srcNamespace, err := request.Payload.String(srcNamespaceCol)
-		if err != nil {
-			log.Printf("Invalid user input, CRD creation or Traceflow request may fail: "+
-				"failed to get srcNamespace as string: %s", err)
-			alert := action.CreateAlert(action.AlertTypeError, fmt.Sprintf("Failed to get source namespace as "+
-				"string: %s", err), action.DefaultAlertExpiration)
-			request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
-			return nil
-		}
-		if errs := validation.ValidateNamespaceName(srcNamespace, false); len(errs) != 0 {
-			log.Printf("Invalid user input, CRD creation or Traceflow request may fail: "+
-				"failed to validate source namespace string %s, errs: %#v", srcNamespace, errs)
-			alert := action.CreateAlert(action.AlertTypeError, fmt.Sprintf("Invalid source namespace string, "+
-				"please check your input and submit again."), action.DefaultAlertExpiration)
-			request.DashboardClient.SendAlert(request.Context(), request.ClientID, alert)
-			return nil
+	case addTfAction, addLiveTfAction:
+		isLiveTraffic := false
+		if actionName == addLiveTfAction {
+			isLiveTraffic = true
 		}
 
-		srcPod, err := request.Payload.String(srcPodCol)
+		srcNamespace, err := request.Payload.String(srcNamespaceCol)
```

Comment:
For live-traffic traceflow, the column can be empty, right? Could you confirm that empty input will not cause any error here? I am asking because that we used to use request.Payload.OptionalString() for optional field.

Reply:
yes, it works with empty source without any error when the destination has been provided, I think it's not a pure optional string because it depends on destination. per may understanding, the OptionalString() should be used for those optional fields under any circumstance.

Reply:
OK, got it. I agree that OptionalString() should be used for those optional fields under any circumstance and just want to make sure empty input gets no errors here. Thanks.